### DISCUSSION TITLE
Anyone able to get prisma working on the new Apple Mac M1 devices?

### DISCUSSION THREAD

**Discussion by ambidtech** [prisma/prisma#4739] [Q&A] [answered]
I read docker support was not ready yet, but docker recently published a preview of their new version that seems to be working.  Next issue is that MySQL images are only available on x86_64 and not arm, but MariaDb has one.  Unfortunately, I'm not versed enough in Docker to be able to get that working.  Finally, I tried installing prisma with a postgres db but couldn't get that working either.

Has anyone had success getting prisma working on the new Apple MC M1 devices?  If so, please advise how you got it working.

Edit: Here is the Error Message when I `prisma init` for a mysql db and run `docker-compose up -d`

"ERROR: no matching manifest for linux/arm64/v8 in the manifest list entries" 

I've come to learn that this is because there isn't a mysql image for arm yet (but there is one for mariadb).

So when I changed the docker-compose file to load the `mariadb` image instead, `docker-compose up -d` runs successfully, but when I open `http://localhost:4466` in my browser localhost refuses to connect.

The same "localhost refused to connect" result also occurs if I set up prisma for postgres or mongodb too.

Any help is appreciated

**Comment by ryands17** [comment, marked as answer]
Hey @ambidtech 👋 
The latest `dev` version of Prisma supports Apple M1 Macs. To get the latest, you would need to run:

```
yarn add @prisma/client@dev
yarn add @prisma/cli@dev -D
```

Then you should be able to connect to any database in that case either local or on the cloud. Let me know if this works :)

**Reply by ambidtech** [reply]
thanks @ryands17!  is this for Prisma v1 and 2, or just Prisma 2?

**Reply by ryands17** [reply]
Just Prisma 2 for now as Prisma 1 will get no new features/enhancements.

Also a thing to note here is that we do not support M1 natively right now, but the intel binaries work without issues on M1 so Prisma should technically work on M1 Macs :)

**Reply by ambidtech** [reply]
thanks again @ryands17!  

Technically, i _think_ prisma 1 worked, but docker couldn't connect to mysql because the mysql image isn't compatible with arm processors.  I tried switching to mariadb (which is compatible) as well as postgres, and did not receive any errors for those, however, after running `docker-compose up -d`, I couldn't access `http://localhost:4466`, so I'm at a loss how to proceed.

Any clues/suggestions would be super appreciated!

**Reply by ryands17** [reply]
If you're getting started, then I would strongly suggest using Prisma 2 and not Prisma 1 at this point as Prisma 2 has a lot of features and enhancements that Prisma 1 doesn't.

> Technically, i think prisma 1 worked, but docker couldn't connect to mysql

Have you tried the same with the latest dev version of Prisma 2?

**Reply by jacobsilver2** [reply]
Exact same thing has been happening to me btw. I also tried MariaDB thinking it might fix this issue but am running into the very same problem you are. I have a prisma 1 project that I've been slacking on moving over to prisma 2...maybe now is the time.

**Reply by AilsonFreire** [reply]
|'m having this issue too. I'm trying to run an old project from my work that uses prisma1 and it can not be updated. It always worked in my previous machine, but right now, using M1 I'm facing the same  `couldn't access http://localhost:4466`

**Reply by camachojuan** [reply]
Was anyone able to make prisma v1 work on M1?

**Reply by khuezy** [reply]
If you guys are having issues still, perhaps this can help: https://stackoverflow.com/questions/68476229/m1-related-prisma-cant-reach-database-server-at-database5432

Basically add `&connect_timeout=300` to your `DATABASE_URL`

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
